src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1388,7 +1421,29 @@ bool AdaptDispatch::ForwardTab(const size_t numTabs)
 // True if handled successfully. False otherwise.
 bool AdaptDispatch::BackwardsTab(const size_t numTabs)
 {
-    return _pConApi->PrivateBackwardsTab(numTabs);
+    CONSOLE_SCREEN_BUFFER_INFOEX csbiex = { 0 };
+    csbiex.cbSize = sizeof(CONSOLE_SCREEN_BUFFER_INFOEX);
+    bool success = _pConApi->GetConsoleScreenBufferInfoEx(csbiex);
+    if (success)
+    {
+        const auto width = csbiex.dwSize.X;
+        const auto row = csbiex.dwCursorPosition.Y;
+        auto column = csbiex.dwCursorPosition.X;
+        auto tabsPerformed = 0u;
+
+        _InitTabStopsForWidth(width);
+        while (column > 0 && tabsPerformed < numTabs)
+        {
+            column--;
+            if (til::at(_tabStopColumns, column))
```

Comment:
we ended up bringing in a `dynamic_bitset`, which looks like `std::vector<bool>` but is somewhat more ergonomic. Is there a clever trick we can use with `dynamic_bitset`'s `find_first()` and `find_next(size_t)` operations here? Perhaps not, because we're walking it in reverse...

Reply:
perhaps for ForwardTab though

Reply:
I've just had a go at using `dynamic_bitset`, and I don't think it's an improvement. The fact that the width boundary has to be special cased makes the code slightly messier, and it introduces the need for narrow cast which is annoying. Also the use of `find_next` triggers an "unreachable code" warning in the header that has to be disabled - I'm not sure why that is.

And after all that, the generated assembly actually looks way worse than the simple `vector` approach. It's possible someone else could make better use of the bitset functionality than I have, but my initial impression is that it doesn't help.